framework/update/helper/updateui_win.cpp

```
@@ -0,0 +1,642 @@
+COLORREF blend(COLORREF from, COLORREF to, int percentOfTo)
+{
+    auto mix = [percentOfTo](int a, int b) {
+        return (a * (100 - percentOfTo) + b * percentOfTo) / 100;
+    };
+
+    return RGB(mix(GetRValue(from), GetRValue(to)),
+               mix(GetGValue(from), GetGValue(to)),
+               mix(GetBValue(from), GetBValue(to)));
+}
+
+bool parseColor(const std::string& text, COLORREF& color)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)